src/commands/generate/app.ts

```diff
@@ -1,84 +1,116 @@
+  static summary = 'Generates a NextJS reference application that integrates Affinidi Login. Requires git'
   static examples = [
     '<%= config.bin %> <%= command.id %>',
-    '<%= config.bin %> <%= command.id %> -p <destination_path>',
-    '<%= config.bin %> <%= command.id %> --path <destination_path> --force',
+    '<%= config.bin %> <%= command.id %> -p "../my-app" -u affinidi',
+    '<%= config.bin %> <%= command.id %> --path "../my-app" --use-case auth0 --force',
   ]
 
   static flags = {
+    'use-case': Flags.string({
+      char: 'u',
+      summary: 'Use case to generate',
+      description: `Use ${chalk.italic('affinidi')} to generate an app that integrates Affinidi Login directly\n\
+      Use ${chalk.italic('auth0')} to generate an app that integrates Affinidi Login through Auth0`,
+      options: Object.values(RefAppUseCases),
+    }),
     path: Flags.string({
       char: 'p',
       summary: 'Relative or absolute path where reference application should be cloned into',
     }),
     force: Flags.boolean({
       summary: 'Override destination directory if exists',
     }),
-    'client-id': Flags.string({
-      summary: 'Affinidi login configurations clientId',
-    }),
-    'client-secret': Flags.string({
-      summary: 'Affinidi login configurations clientSecret',
-    }),
-    'access-token': Flags.string({
-      summary: 'IDP access token',
-    }),
-    domain: Flags.string({
-      summary: 'Tenant domain',
-    }),
   }
 
   public async run(): Promise<void> {
     const { flags } = await this.parse(GenerateApp)
+    if (flags['no-input'] && !flags['use-case']) {
+      throw new CLIError(giveFlagInputErrorMessage('use-case'))
+    }
+    const useCase =
+      flags['use-case'] ??
+      (await select({
+        message: `Select the app to generate. ${chalk.italic('affinidi')} for basic Affinidi Login, ${chalk.italic(
+          'auth0',
+        )} for Affinidi Login through Auth0`,
+        choices: Object.values(RefAppUseCases).map((value) => ({
+          name: value,
+          value,
+        })),
+      }))
     const promptFlags = await promptRequiredParameters(['path'], flags)
 
     ux.action.start('Generating reference application')
-    await cloneWithDegit(GIT_URL, promptFlags.path, flags.force)
+    await cloneWithDegit(`affinidi/reference-app-affinidi-vault/use-cases/${useCase}`, promptFlags.path, flags.force)
     ux.action.stop('Generated successfully!')
 
-    if (flags['no-input']) {
-      if (!flags['client-id']) {
-        throw new CLIError(giveFlagInputErrorMessage('client-id'))
-      }
-      if (!flags['client-secret']) {
-        throw new CLIError(giveFlagInputErrorMessage('client-secret'))
-      }
-      if (!flags['access-token']) {
-        throw new CLIError(giveFlagInputErrorMessage('access-token'))
-      }
-      if (!flags.domain) {
-        throw new CLIError(giveFlagInputErrorMessage('domain'))
+    if (!flags['no-input']) {
```

Comment:
What will happen in the case of flags['no-input']?

Reply:
It will skip all of the configuration block and just clone the app. Configuring the app is very dependent on the use case and requires specific input. I initially had configuration flags and their description was already getting complicated with just two apps, so I decided to remove them. As we add more and more apps that would make this command very hard to understand.  If we really need the configuration part to be supported via flags I suggest to create a command to configure each app type.

Reply:
Okay, wanted to make sure it's a conscious choice.